src/model/client.ts

```diff
@@ -26,9 +26,25 @@ export class CommandoClient extends Client {
 
     this.options = options
 
-    this.on('message', async (msg: Message) => {
-      await this.onMessage(msg)
-    })
+    this.on('message', this.onMessage)
+  }
+
+  /**
+   * Calls a callback when a message containing a command is processed.
+   * @param callback The function to call on the message
+   */
+  public onCommandMessage(
+    callback: (msg: CommandoMessage, cmd: Command, prefix: string) => void
+  ): this {
```

Comment:
Callback functions should have a return type of `void`.

The implementation of the function should look like:
```ts
client.onCommandMessage((msg, cmd, prefix) => doSomething());
```